engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -109,6 +112,95 @@ public List<Endpoint> getTargetEmailSubscriptionEndpoints(String orgId, String b
         return endpoints;
     }
 
+    /**
+     * Increments the server errors counter of the endpoint identified by the given ID.
+     * @param endpointId the endpoint ID
+     * @param maxServerErrors the maximum server errors allowed from the configuration
+     * @return {@code true} if the endpoint was disabled by this method, {@code false} otherwise
+     */
+    @Transactional
+    public boolean incrementEndpointServerErrors(UUID endpointId, int maxServerErrors) {
+        /*
+         * This method must be an atomic operation from a DB perspective. Otherwise, we could send multiple email
+         * notifications about the same disabled endpoint in case of failures happening on concurrent threads or pods.
+         */
+        Optional<Endpoint> endpoint = lockEndpoint(endpointId);
+        /*
+         * The endpoint should always be present unless it's been deleted recently from another thread or pod.
+         * It may or may not have been disabled already from the frontend or because of a 4xx error.
+         */
+        if (endpoint.isPresent() && endpoint.get().isEnabled()) {
+            if (endpoint.get().getServerErrors() + 1 > maxServerErrors) {
+                /*
+                 * The endpoint exceeded the max server errors allowed from configuration.
+                 * It is therefore disabled and the errors counter is reset.
```

Comment:
I was wondering if we should leave the max server errors to verify that it was auto-disabled because of server errors.

We could reset this value when updating the integration - and that would make sense I guess - if we update the URL or any configuration, the response might be different.

Reply:
We can verify the reason why it was disabled from the log entries in `WebhookTypeProcessor`.

I'd rather not reset this field from another place like `backend` REST APIs as it will make the whole thing more complex and harder to understand/maintain.

Reply:
Not exaclty sure I like this behavior but we can always revisit this - maybe it's an unlikely scenario - besides if it keeps failing it should be disabled.

Reply:
If you don't like it then it makes me want to improve it 😄 I'll submit another PR with something better.